tests/test_action.py

```diff
@@ -45,3 +46,25 @@ class TestAction(Action):
 
     with pytest.raises(TypeError):
         TestAction()
+
+
+def test_accept_method_calls_visit_and_response_and_requests_accept():
+    mock_wrapper = Mock()
+    mock_wrapper.mocked_visitor = MagicMock()
+    mock_wrapper.mocked_request = MagicMock()
+    mock_wrapper.mocked_response1 = MagicMock()
+    mock_wrapper.mocked_response2 = MagicMock()
+
+    action = Action(
+        name='Test Action',
+        request=mock_wrapper.mocked_request,
+        responses=[mock_wrapper.mocked_response1, mock_wrapper.mocked_response2]
+    )
+    action.accept(mock_wrapper.mocked_visitor)
+
+    assert mock_wrapper.mock_calls[-4:] == [
```

Comment:
This mock_calls attribute from wrapper is used to check methods call order.

Reply:
whats is the reason to slice?

Reply:
ah, because `[0]` is the `Mock()`

Reply:
In this case, we are checking for the last 4 calls because of the objects handling on initialization. In other files this doesn't happen.